### DISCUSSION TITLE
Auth first time (new user)

### DISCUSSION THREAD

**Discussion by zakaria-chahboun** [supabase/supabase#3395] [Questions] [answered]
Hi Supabasers :P 
How i can know if a user is Logged In for the first time (a new user)?

Thanks!

**Comment by dthyresson** [comment, marked as answer]
> How i can know if a user is Logged In for the first time (a new user)?

HI @zakaria-chahboun do you mean that they "signed up" or the first login subsequent to a sign up?

For example, I signed up to your app last Tuesday, but I did not come back until this morning and I signed in. 

Is the "first" event the signup (Tuesday) or today?


If it is the former - ie Sign Up - then a trigger can help here.

See: https://github.com/supabase/supabase/discussions/306#discussioncomment-138424


```sql
/**
* This trigger automatically creates a user entry when a new user signs up via Supabase Auth.
*/ 
create function public.handle_new_user() 
returns trigger as $$
begin
  insert into public.users (id)
  values (new.id);
  return new;
end;
$$ language plpgsql security definer;
create trigger on_auth_user_created
  after insert on auth.users
  for each row execute procedure public.handle_new_user();
```

Instead of adding a new row to the public users table, you could invoke something else -- using the created_at or confirmed_at values to know when they signed up.

**Reply by zakaria-chahboun** [reply]
i mean something in the metadata, just to show him a welcome message!
i guess i can handle this by using `confirmed_at` + `created_at`, like you said!
